helix-core/src/auto_pairs.rs

```diff
@@ -120,15 +120,21 @@ impl Default for AutoPairs {
 //   middle of triple quotes, and more exotic pairs like Jinja's {% %}
 
 #[must_use]
-pub fn hook(doc: &Rope, selection: &Selection, ch: char, pairs: &AutoPairs) -> Option<Transaction> {
+pub fn hook(
+    doc: &Rope,
+    selection: &Selection,
+    ch: char,
+    pairs: &AutoPairs,
+    overtype: bool,
+) -> Option<Transaction> {
     log::trace!("autopairs hook selection: {:#?}", selection);
 
     if let Some(pair) = pairs.get(ch) {
         if pair.same() {
             return Some(handle_same(doc, selection, pair));
         } else if pair.open == ch {
             return Some(handle_open(doc, selection, pair));
-        } else if pair.close == ch {
+        } else if pair.close == ch && overtype {
```

Comment:
Shouldn't the flag be handled inside `handle_close`?

Reply:
I thought it might make more sense to just return and insert the closing pair instead of creating a new transaction and then checking. It definitely makes more sense inside though.